src/ini.zig

```diff
@@ -0,0 +1,1076 @@
+                        };
+                        break :brk &install.default_registry.?;
+                    };
+
+                    switch (conf_item.optname) {
+                        ._authToken => v.token = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                        .username => v.username = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                        ._password => v.password = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                        ._auth, .email, .certfile, .keyfile => unreachable,
+                    }
+                    continue;
+                }
+
+                var matched_at_least_one = false;
+                for (registry_map.scopes.keys(), registry_map.scopes.values()) |*k, *v| {
+                    const url = url_map.get(k.*) orelse unreachable;
+
+                    if (std.mem.eql(u8, bun.strings.withoutTrailingSlash(url.host), bun.strings.withoutTrailingSlash(conf_item_url.host))) {
+                        if (conf_item_url.hostname.len > 0) {
+                            if (!std.mem.eql(u8, bun.strings.withoutTrailingSlash(url.hostname), bun.strings.withoutTrailingSlash(conf_item_url.hostname))) {
+                                continue;
+                            }
+                        }
+                        matched_at_least_one = true;
+                        switch (conf_item.optname) {
+                            ._authToken => v.token = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                            .username => v.username = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                            ._password => v.password = allocator.dupe(u8, conf_item.value) catch bun.outOfMemory(),
+                            ._auth, .email, .certfile, .keyfile => unreachable,
+                        }
+                        // We have to keep going as it could match multiple scopes
+                        continue;
+                    }
+                }
+
+                if (!matched_at_least_one) {
+                    Output.warn(
+                        "The follwing .npmrc registry option was not applied:\n\n  <b>{s}<r>\n\nBecause we couldn't find the registry: <b>{s}<r>.",
+                        .{
+                            conf_item,
+                            conf_item.registry_url,
+                        },
+                    );
+                }
+            }
+        }
+    }
+
+    if (log.hasErrors()) {
+        return error.ParserError;
+    }
+
+    if (!bun.Environment.isDebug) {
+        if (!@import("./bun.js/module_loader.zig").ModuleLoader.is_allowed_to_use_internal_testing_apis)
+            return;
+    }
+
+    if (bun.getenvTruthy("BUN_TEST_LOG_DEFAULT_REGISTRY")) {
+        if (install.default_registry) |reg| {
+            Output.print("Default registry url: {s}\n", .{reg.url});
```

Comment:
Per discussion IRL, let's add a test-only wrapper function that can be called from JS

Then if there are any log messages, it calls `.toJS` on it and returns or throws that.

Separately, lets enable `bun install --verbose` and check the output to verify that the expected URL is requested